switch/switch_main.ml

```diff
@@ -68,7 +80,27 @@ module Config = struct
       let doc = "Directory containing state files" in
       Arg.(value & opt (some string) default.statedir & info ["statedir"] ~doc)
     in
-    Term.(pure make $ path $ pidfile $ configfile $ statedir)
+    let trace_entries =
+      let doc =
+        "Maximum number of trace entries buffer (for message-cli tail)"
+      in
+      Arg.(value & opt (some int) None & info ["trace-entries"] ~doc)
```

Comment:
I would have expected `opt int default_trace_entries`: when no value is provided, the default is used. This also makes the default being shown in `--help`.

Reply:
This is configurable both in the config file and on the cmdline, hence the need for None, otherwise we'd always override the value from the config file with the one from the cmdline. But maybe that is overkill: just having it configurable on the cmdline should suffice (one can edit the systemd service file if they have to, this is a debugging-only flag after all). I'll simplify the code a bit.

Reply:
I only take the cmdline value into consideration now, and I moved the cmdline definition to Traceext.term and Traceext.config.
This way the trace configuration is local to its own module (so it is easier to add more in the future without modifying the main module)